### PR TITLE
Extract Markdown parse/produce as public pkg/markdown library

### PR DESCRIPTION
**Pull request by jeduden**
Completes plan #163 by extracting mdsmith's Markdown parsing and production capabilities into a public Go library at `pkg/markdown`.

## Summary

This PR establishes `pkg/markdown` as mdsmith's canonical, importable Markdown surface. It moves parsing logic, processing instruction handling, and frontmatter utilities from `internal/lint` into the public package, making them available to external consumers while maintaining a single goldmark parser configuration across the codebase.

## Key Changes

- **New public package `pkg/markdown`**: Exports the canonical goldmark parser and Markdown processing utilities
  - `parser.go`: `NewParser()` returns the shared parser instance with CommonMark + processing instruction support
  - `parse.go`: `Parse()` and `Document` type for parsing Markdown with frontmatter separation
  - `pi.go` & `pi_parser.go`: Processing instruction (`<?...?>`) AST node and parser
  - `frontmatter.go`: `StripFrontMatter()` utility for YAML frontmatter handling
  - `produce.go`: `Edit` and `Splice()` for Markdown mutation
  - `doc.go`: Package documentation and compatibility policy

- **Refactored `internal/lint`**: Now imports from `pkg/markdown` instead of duplicating logic
  - `pi.go`, `pi_parser.go`, `frontmatter.go` reduced to thin wrappers or removed
  - `file.go` and `lint_coverage_test.go` updated to use public APIs
  - `pi_test.go` moved to `pkg/markdown/pi_test.go` with expanded coverage

- **Updated internal consumers**: `internal/release/syncdocs.go` and `internal/rules/markdownflavor/parser.go` now import from `pkg/markdown`

- **Documentation**: Added `docs/development/markdown-library.md` describing the public API, compatibility policy, and usage

- **Architecture updates**: Updated architecture docs to reflect `pkg/markdown` as the single Markdown surface boundary

- **Test infrastructure**: Added golden-file test for `syncdocs` reconciliation and comprehensive unit tests for all public APIs

- **Plan status**: Marked plan #163 as complete (✅)

## Implementation Details

- Parser is cached via `sync.Once` to ensure a single instance across the application
- Processing instructions are registered as a custom goldmark block parser
- Frontmatter stripping is idempotent and handles edge cases (empty files, missing delimiters)
- All public APIs include unit tests with 100% coverage expectations
- Backward compatibility maintained: internal code continues to work via public imports

https://claude.ai/code/session_01RBQSS4X7e6mfrkyXmDzGRX

### CODE REVIEW

File: docs/development/markdown-library.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+- The `<?…?>` marker grammar is a shipped
+  [generated-section](../background/concepts/generated-section.md)
+  contract. Once parsed a certain way, it keeps
+  parsing that way until the next major. A
+  deprecation diagnostic may precede removal.
+- `Splice` output is byte-exact. For the same
+  input it must keep producing the same bytes.
+  The sync-docs golden corpus
+  (`internal/release/testdata/reconcile_corpus.golden`)
+  pins this across the whole docs tree.
```
[the remark this answers is not in the document]
Comment (jeduden) [reply docs/development/markdown-library.md]: Addressed in 7779fb4, but differently than suggested. The `reconcile_corpus.golden` reference and the entire per-case `internal/release/testdata/reconcile/` tree were removed, not repointed: they duplicated the pre-existing inline `TestReconcileDocForHugo_*` table tests in `internal/release/syncdocs_test.go`, which already cover every branch. The compatibility-policy bullet now points at those tests plus `pkg/markdown`'s `TestSplice` cases, so the stale reference is gone and no `testdata/reconcile/` path is referenced by design.
